adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -235,7 +235,23 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
     }
   }
 
-  def loadBam(filePath: String): RDD[AlignmentRecord] = {
+  /**
+   * Loads a SAM/BAM file.
+   *
+   * This reads the sequence and record group dictionaries from the SAM/BAM file
+   * header. SAMRecords are read from the file and converted to the
+   * AlignmentRecord schema.
+   *
+   * @param filePath Path to the file on disk.
+   *
+   * @return Returns a Tuple3 containing (an RDD of reads, the sequence
+   *   dictionary representing the contigs these reads are aligned to
+   *   if the reads are aligned, the record group dictionary for the reads
+   *   if one is available).
+   *
+   * @see loadAlignments
+   */
+  def loadBam(filePath: String): (RDD[AlignmentRecord], SequenceDictionary, RecordGroupDictionary) = {
```

Comment:
> I share the opinion from the Google Guava folks that tuple types are awful obfuscators. It is off-putting for non-Scala-natives and makes for a weak API. What if the number or order of values in the tuple changes over time? Could the tuple be instead something with a name?

Thanks for suggesting this. I think you've got a good point, and I think it'd be a straightforward refactor to support. I'll add a commit that proposes a better interface.

> In either case I'd still like to keep the RDD API, e.g.

I'm opposed to your proposal for the reasons I gave above (not strongly, I'm something around a -0 to -0.5) but I think the two of us are kind of going in circles here. Perhaps let's wait and get more comments from folks (CC @laserson @massie @tdanford @ryan-williams) and see what the consensus is. If